src/runtime/omp/omp_queue.cpp

```diff
@@ -69,8 +71,100 @@ bool is_contigous(id<3> offset, range<3> r, range<3> allocation_shape) {
   return true;
 }
 
+class instrumentation_task_guard;
+
+template <class BaseInstrumentation>
+class omp_task_timestamp : public BaseInstrumentation {
+public:
+  friend class instrumentation_task_guard;
+
+  virtual profiler_clock::time_point get_time_point() const override {
+    return _time;
```

Comment:
This can still be misused by calling `get_time_point()` before `wait()`, which will race on `_time`.

Reply:
Not really - the whole point of the instrumentation API is that `instrumentation_set` guarantees that no instrumentation objects will be handed out until `wait()` has returned. This simplifies the design of instrumentations because it can be assumed that they will only be accessed (after initial setup) once they are ready.

Reply:
Ok, then I'll be happy with an `assert(_signal.has_signalled())` in there.